### DISCUSSION TITLE
vm.wrap and vm.roll

### DISCUSSION THREAD

**Discussion by yasharyan0904** [Cyfrin/foundry-full-course-cu#1876] [Q&A]
Can anyone please explain the exact working of vm.wrap(manipulate the timestamp of the current block, simulating the passage of time) and vm.roll in terms of blocks on a blockchain, like what it individually does?

code
```
vm.wrap(block.timestamp + interval + 1);
vm.roll(block.number + 1);
```

**Comment by EngrPips** [comment]
Hello @yasharyan0904, So `vm.warp` increases a blockchain's `timestamp` (takes us to the future) while `vm.roll` increases the number of blocks mined for a blockchain. These are cheat codes in foundry to simulate time passage and block increment. Please don't hesitate to ask questions if anything needs clarification.

**Reply by yasharyan0904** [reply]
vm.roll increses the no. of blocks to cope up with the timestamp of the vm.wrap??

vm.roll(block.number + 1); Here the block number is the block number after mining for the required timestamp?

**Reply by EngrPips** [reply]
I am unsure of the intricate details of how everything works together to make sense of it, but it makes sense that if some time has passed, some blocks must have been mined.  If anything is still confusing, please ask questions.
